store/query.go

```diff
@@ -289,19 +290,26 @@ func (r *Result) ToMap() (data map[string]interface{}) {
 	data = make(map[string]interface{})
 	data["id"] = r.Id
 	data["kind"] = r.Kind
-	var ts int64
+	var ts_latest int64
+	var ts_oldest int64
+	ts_oldest = time.Now().UnixNano()
 	for pred, versions := range r.Columns {
 		// During conversion to JSON, to keep things simple,
 		// we're dropping older versions of predicates, and
 		// source and ts information across all the predicates,
 		// keeping only the latest one.
 		data[pred] = versions.Latest().Value
-		data["source"] = versions.Latest().Source // Loss of information.
-		if versions.Latest().NanoTs > ts {
-			ts = versions.Latest().NanoTs
+		if versions.Latest().NanoTs > ts_latest {
+			ts_latest = versions.Latest().NanoTs
+			data["source"] = versions.Latest().Source // Loss of information.
```

Comment:
Drop the "source", call it data["modifier"]. So, we'll have `modifier` and `creator`.